iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -351,7 +345,7 @@ inline typename vector<T, Capacity>::const_iterator vector<T, Capacity>::end() c
 {
     // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-    return reinterpret_cast<const_iterator>(&(at_unchecked(0)) + m_size);
+    return reinterpret_cast<const_iterator>(&at_unchecked(m_size));
```

Comment:
When using `at_unchecked(m_size)`, `m_data[m_size]` is accessed which is undefined behavior. Shouldn't we keep the pointer arithmetic so that there is only undefined behavior when the user attempts to access the element (which is outside of the vector)?

Reply:
@FerdinandSpitzschnueffler I have reverted the change and suppressed the warning with the justification that the template parameter ensures predictability in the memory layout therefore the correctness of pointer arithmetic.

cc. @elBoberido